### ISSUE TITLE
ValueError: dictionary update sequence element #0 has length 1; 2 is required

### ISSUE THREAD

**Issue by SanskritFritz** [tokland/youtube-upload#2]
```youtube-upload --title="Check" Check.avi```

```
Using client secrets: /usr/share/youtube_upload/client_secrets.json
Using credentials file: /home/frank/.youtube-upload-credentials.json
Traceback (most recent call last):
  File "/usr/bin/youtube-upload", line 164, in <module>
    sys.exit(lib.catch_exceptions(EXIT_CODES, main, sys.argv[1:]))
  File "/usr/lib/python2.7/site-packages/youtube_upload/lib.py", line 21, in catch_exceptions
    fun(*args, **kwargs)
  File "/usr/bin/youtube-upload", line 161, in main
    run_main(parser, options, args)
  File "/usr/bin/youtube-upload", line 124, in run_main
    video_id = upload_video(youtube, options, video_path, len(args), index)
  File "/usr/bin/youtube-upload", line 94, in upload_video
    "location": lib.string_to_dict(options.location),
  File "/usr/lib/python2.7/site-packages/youtube_upload/lib.py", line 38, in string_to_dict
    return dict(pair.split("=") for pair in pairs)
ValueError: dictionary update sequence element #0 has length 1; 2 is required
```

**Comment by dbrandt** [comment]
I hit this problem today. Taking a quick look at the code you should be able to just pass `--location="="` to get past this, but it's obviously a bug in need of fixing.

I'd much rather see the whole argument parsing code redone with the `argparse` module since it's the future way of doing things and much more powerful than optparse.

**Comment by ManuCart** [comment]
Same for me this morning
Fixed with ````--location="="````

**Comment by tokland** [comment]
Sorry, guys, I was getting no notifications on issues. Fixed: 07bcc287
